code/datums/chatmessage.dm

```diff
@@ -168,69 +230,145 @@
+					//was still in the second stage of animation. if the time was wrong and the client was in the third stage of animation when the updated step
+					//arrives, then this will look weird.
+					animate(other_message_image, alpha = 0, time = CHAT_MESSAGE_EOL_FADE, flags = ANIMATION_PARALLEL)
+					animate(layer = MESSAGE_ANIMATION_FORCE_FADE_LAYER_MARK, time = 0)
+
+					preexisting_message.fade_times_by_image[other_message_image] = world.time //make sure we can tell afterwards if its already fading
+
+			//make it move upwards
+			animate(other_message_image, pixel_y = other_message_image.pixel_y + mheight, time = CHAT_MESSAGE_SPAWN_TIME, flags = ANIMATION_PARALLEL)
+
+	var/maptext_used = MAPTEXT(complete_text)
+	var/maptext_x_used = (CHAT_MESSAGE_WIDTH - owner.bound_width) * -0.5
+
+	var/image/message = create_new_image(target, maptext_used, mheight, maptext_x_used)
+
+	//handle the client side animations for the image
 	animate(message, alpha = 255, time = CHAT_MESSAGE_SPAWN_TIME)
+	animate(alpha = 255, time = lifespan)
+	animate(alpha = 0, time = CHAT_MESSAGE_EOL_FADE)
+	animate(layer = MESSAGE_ANIMATION_DEFAULT_LAYER_MARK, time = 0)
+	//client wont see the results of this last step since it wont be visible. its just used to mark to the server whether the last animation stage was forced
+
+	handle_new_image_association(message, owned_by, our_approx_lines)
+
+/datum/chatmessage/proc/create_new_image(atom/target, maptext, mheight, maptext_x)
+	var/static/mutable_appearance/template = create_runechat_template()
+	template.layer = CHAT_LAYER + CHAT_LAYER_Z_STEP * current_z_idx
+	template.maptext_height = mheight
+	template.pixel_y = target.maptext_height
+	template.pixel_x = (target.maptext_width * 0.5) - 16
+	template.maptext_x = maptext_x
+	template.maptext = maptext
 
-	// Register with the runechat SS to handle EOL and destruction
-	var/duration = lifespan - CHAT_MESSAGE_EOL_FADE
-	fadertimer = addtimer(CALLBACK(src, .proc/end_of_life), duration, TIMER_STOPPABLE|TIMER_DELETE_ME, SSrunechat)
+	// Build message image
+	var/image/message = image(loc = message_loc)
+	message.appearance = template.appearance
 
-/**
- * Applies final animations to overlay CHAT_MESSAGE_EOL_FADE deciseconds prior to message deletion,
- * sets timer for scheduling deletion
- *
- * Arguments:
- * * fadetime - The amount of time to animate the message's fadeout for
- */
-/datum/chatmessage/proc/end_of_life(fadetime = CHAT_MESSAGE_EOL_FADE)
-	isFading = TRUE
-	animate(message, alpha = 0, time = fadetime, flags = ANIMATION_PARALLEL)
-	addtimer(CALLBACK(GLOBAL_PROC, /proc/qdel, src), fadetime, TIMER_DELETE_ME, SSrunechat)
+	return message
+
+/proc/create_runechat_template()
+	var/mutable_appearance/template = new()
+	template.maptext_width = CHAT_MESSAGE_WIDTH
+	template.plane = RUNECHAT_PLANE
+	template.appearance_flags = APPEARANCE_UI_IGNORE_ALPHA | KEEP_APART
+	template.alpha = 0
+	template.maptext_width = CHAT_MESSAGE_WIDTH
```

Comment:
could this just be a subtype? or is that too cursed

Reply:
i tried that and it didnt work for some reason, might be due to using an MA